Simplified/NYPLAnnouncementManager.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+private let announcementKey: String = "NYPLPresentedAnnouncementKey"
+private let filename: String = "NYPLPresentedAnnouncementsList"
+
+class NYPLAnnouncementManager {
+  private static var presentedAnnouncement: Set<String> = Set<String>()
+    
+  class func presentAnnouncements(_ announcements: [Announcement]) {
+    for announcement in announcements {
+      if shouldPresentAnnouncement(id: announcement.id) {
+        DispatchQueue.main.async {
+          let vc = NYPLAnnouncementViewController(announcement: announcement)
+          NYPLRootTabBarController.shared()?.safelyPresentViewController(vc, animated: true, completion: nil)
+        }
+      }
+    }
+  }
+  
+  // MARK: - Read
+    
+  class func shouldPresentAnnouncement(id: String) -> Bool {
+    guard let filePathURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first?.appendingPathComponent(filename),
+      let filePathData = try? Data(contentsOf: filePathURL),
+      let unarchived = try? NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(filePathData),
+      let presented = unarchived as? Set<String> else {
+        return true
+    }
+    presentedAnnouncement = presented
+    
+    return !presentedAnnouncement.contains(id)
+  }
+  
+  // MARK: - Write
+    
+  class func addPresentedAnnouncement(id: String) {
+    presentedAnnouncement.insert(id)
+    
+    storePresentedAnnouncementToFile()
+  }
+
+  // For testing use
```

Comment:
if it's for testing only I would move this into an extension. A swift sort-of pattern is to do:
```
extension YourClass {
  func testing_deletePresentedAnnouncement() {
  ...
  }
}
```